src/dependencies.jl

```diff
@@ -263,7 +263,7 @@ provider(::Type{Binaries},uri::URI; opts...) = RemoteBinaries(uri)
 provider(::Type{Binaries},path::AbstractString; opts...) = CustomPathBinaries(path)
 provider(::Type{SimpleBuild},steps; opts...) = SimpleBuild(steps)
 provider{T<:BuildProcess}(::Type{BuildProcess},p::T; opts...) = provider(T,p; opts...)
-provider(::Type{BuildProcess},steps::Union(BuildStep,SynchronousStepCollection); opts...) = provider(SimpleBuild,steps; opts...)
+@compat provider(::Type{BuildProcess},steps::Union{BuildStep,SynchronousStepCollection}; opts...) = provider(SimpleBuild,steps; opts...)
```

Comment:
And for this one it should be less problematic (since the rules have been in for longer) and I'm not really sure how can this be reasonably tested. This is why I'd like to tag a new version and fix it if people complains..... ;-p